codalib/util.py

```diff
@@ -21,12 +21,100 @@
 import subprocess
 
 # not really thrilled about duplicating these globals here -- maybe define them in coda.bagatom?
-PREMIS_NAMESPACE = "http://www.loc.gov/standards/premis/v1"
+PREMIS_NAMESPACE = "info:lc/xmlns/premis-v2"
 PREMIS = "{%s}" % PREMIS_NAMESPACE
 PREMIS_NSMAP = {"premis": PREMIS_NAMESPACE}
-dateFormat = "%Y-%m-%d %H:%M:%S"
 svn_version_path = "/usr/bin/svnversion"
 
+# constants for time parsing/formatting
+XSDT_FMT = "%Y-%m-%dT%H:%M:%S" # this is a stub
+XSDT_TZ_OFFSET = 19 # this should never change
+
+class InvalidXSDateTime(Exception):
+    pass
+
+
+class XSDateTimezone(tzinfo):
+    """ 
+    Concrete subclass of tzinfo for making sense of timezone offsets.
+    Not really worried about localization here, just +/-HHMM
+    """
+
+    def __init__(self, hours=0, minutes=0, sign=1):
+        self.minutes = hours*60+minutes
+        self.minutes *= sign
+
+    def utcoffset(self, dt):
```

Comment:
It's part of the tzinfo interface. tzinfo is an abstract class and it takes the dt arg, so all concrete classes inheriting from it have to include it.